### ISSUE TITLE
Snapshot agent unable to retrieve license in 1.10.x or 1.11.x with snapshot-agent when ACL token is passed via environment variable 

### ISSUE THREAD

**Issue by TJM** [hashicorp/consul#12112]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request. Searching for pre-existing feature requests helps us consolidate datapoints for identical requirements into a single place, thank you!
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.

---

### Overview of the Issue

When using the snapshot agent (enterprise only) and manageSystemACLs: true,  the snapshot agent is unable to retrieve its license and thus is unable to start.

### Reproduction Steps

When running `helm upgrade --install consul hashicorp/consul -f myvalues.yaml` it never completes because the snapshot agent is never able to start.

* myvalues.yaml

```yaml
global:
  acls:
    manageSystemACLs: true
  enterpriseLicense:
    enableLicenseAutoload: true
    secretKey: license-key
    secretName: consul-enterprise-license
client:
  snapshotAgent:
    configSecret:
      secretKey: agent_config
      secretName: consul-snapshot-agent
    enabled: true
```

You will need the requisite secrets for `consul-enterprise-license` and `consul-snapshot-agent`.

### Logs

Note, we are using terraform to apply this helm chart, and have set the timeout to 10m... after 10m, it fails with a timeout error:

```
│ Error: timed out waiting for the condition
│ 
│   with helm_release.consul,
│   on gke_consul.tf line 320, in resource "helm_release" "consul":
│  320: resource "helm_release" "consul" {
│ 
```

### Expected behavior

It should upgrade or install without issues.

### Environment details

If not already included, please provide the following:
- `consul-k8s` version: (default?)
- chart version: 0.39.0
- `values.yaml` used to deploy the helm chart: (see above)

Additionally, please provide details regarding the Kubernetes Infrastructure, as shown below:
- Kubernetes version: v1.22.x 
- Cloud Provider (If self-hosted, the Kubernetes provider utilized): EKS, AKS, GKE, OpenShift (and version), Rancher (and version), TKGI (and version): GKE 1.20.10-gke.1600
- Networking CNI plugin in use: Calico, Cilium, NSX-T Calico?

Any other information you can provide about the environment/deployment.

### Additional Context

See PR hashicorp/consul-k8s#940 for potential workaround.

**Comment by Amier3** [comment]
Giving a quick update on this, we have an open PR we're actively working on for this issue. Thanks you for your patience!

**Comment by TJM** [comment]
FYI: If you need to move to 1.10.x (or 1.11.x) before this is fixed, we have created a workaround. See PR https://github.com/hashicorp/consul-k8s/pull/940 (it is not supported and use at your own risk, but it works for us)

**Comment by david-yu** [comment]
Hi @TJM we will be releasing a fix for this issue towards the end of this month. I'll go ahead and close for now. Please re-open if you find that it is not fixed when trying out patch releases on 1.10.x, 1.11.x, or 1.12.x. Thank you for your patience.
